Added support for setting form type options on parent form (edit, new, form levels)

ogizanagi opened the pull request:
Follows #323.

This PR will allow to set options in order to configure the parent form. For instance : 

```php
entities:
    TestEntity:
        class: AppBundle\Entity\TestEntity
        form:
            form_options: { validation_groups: ['Default', 'my_validation_group'] }
```
will create the form using the following options:
```php
array(
    'validation_groups' => array('Default', 'my_validation_group'),
    'data_class' => 'AppBundle\Entity\TestEntity',
    'attr' => array('class' => '...', 'id' => '...'),
)))
```

Do you think `form_options` is more appropriated here, or `type_options` should be used to be consistent ?

Pierstoval commented:
Sounds good to me to use `form_options` here, and `form_field_option` in the fields configuration

ogizanagi commented:
I added a functional test adding `validation_groups` and `novalidate` html form attribute this morning, but I reverted it as the `FormDataCollector` I used in order to retrieve form validation errors isn't available with symfony <2.4.
I've re-added it with an alternative for 2.3 only. In case we drop the support one day, we'll just have to get rid of the `try/catch` block.
Tests should be ok now.

ogizanagi commented:
Rebased on master, since Tests structure has changed recently.

Pierstoval commented:
So maybe a small rebase to check everything is correct, followed by a merge is good?
@javiereguiluz I know you have plenty of work but I think this feature and #323 are quite awaited...

ogizanagi commented:
Phew! Accidentally removed the functional tests because I made those changes on another computer and didn't retrieve my last changes before push force... 
Github API saved me. As it might help one day : `https://api.github.com/repos/<user>/<repo>/events` (Allow to retrieve almost any commit)

javiereguiluz commented:
@Pierstoval the only blocking issue is that I'm still thinking about this and the other realted PR to see what's the best way to implement this. But I'll make some decission soon.

javiereguiluz commented:
@ogizanagi I'm sorry it took me so long to merge your pull request. I'm glad this feature is now available. Thanks a lot for working on this.

javiereguiluz commented:
Minor changes and documented this option in #481

ogizanagi commented:
Great ! Thank to you @javiereguiluz ! :)
